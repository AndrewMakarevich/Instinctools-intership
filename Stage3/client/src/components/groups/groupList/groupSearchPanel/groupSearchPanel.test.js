import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useState } from 'react';
import GroupSearchPanel from './groupSearchPanel';

let filterParams = {
  page: 2,
  limit: 2,
  filterObject: {
    groupName: 'FirstGroup',
    groupTitle: '',
  },
};

afterEach(() => {
  jest.clearAllMocks();
});

const user = userEvent.setup();

describe("Correct group's search panel", () => {
  test('render', async () => {
    render(
      <GroupSearchPanel
        paramsMap={['groupName', 'groupTitle']}
        queryParams={filterParams}
      />
    );
    expect(screen.getByTestId('search-panel')).toBeInTheDocument();
    expect(screen.getByTestId('search-panel')).toMatchSnapshot();
    expect(
      screen.getByPlaceholderText('Search by groupName')
    ).toBeInTheDocument();
    expect(
      screen.getByPlaceholderText('Search by groupTitle')
    ).toBeInTheDocument();
  });

  test('sending request', async () => {
    const fetchGroups = jest.fn();
    render(
      <GroupSearchPanel
        paramsMap={['groupName', 'groupTitle']}
        queryParams={filterParams}
        fetchGroups={fetchGroups}
      />
    );
    const searchInputs = screen.getAllByTestId('search-input');
    await user.type(searchInputs[0], ' ');

    expect(fetchGroups.mock.calls.length).toBe(1);
    expect(fetchGroups.mock.calls[0][0].page).toBe(1);
  });

  test('work of clear search panel button', async () => {
    const fetchGroups = jest.fn();
    render(
      <GroupSearchPanel
        paramsMap={['groupName', 'groupTitle']}
        queryParams={filterParams}
        fetchGroups={fetchGroups}
      />
    );

    const clearPanelBtn = screen.getByTestId('clear-group-search-panel-btn');
    await user.click(clearPanelBtn);

    expect(fetchGroups.mock.calls.length).toBe(1);
    const filterObjValues = Object.values(
      fetchGroups.mock.calls[0][0].filterObject
    );
    expect(filterObjValues.some((val) => val !== '')).toBe(false);
  });
});
