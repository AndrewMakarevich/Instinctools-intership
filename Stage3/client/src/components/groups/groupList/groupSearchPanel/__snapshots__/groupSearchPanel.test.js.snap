// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct group's search panel render 1`] = `
<section
  class="search-panel"
  data-testid="search-panel"
>
  <div
    class="search-panel__inputs"
  >
    <label
      class="label"
      data-testid="search-input"
      for="search-input"
    >
      <input
        class="input PropTypes.string"
        id="search-input"
        placeholder="Search by groupName"
        value="FirstGroup"
      />
      <span
        class="icon"
      >
        <svg
          class="icon "
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </label>
    <label
      class="label"
      data-testid="search-input"
      for="search-input"
    >
      <input
        class="input PropTypes.string"
        id="search-input"
        placeholder="Search by groupTitle"
        value=""
      />
      <span
        class="icon"
      >
        <svg
          class="icon "
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </span>
    </label>
  </div>
  <button
    class="my-button "
    data-testid="clear-group-search-panel-btn"
  >
    Clear search inputs
  </button>
</section>
`;
